src/controllers/UsersController.php

```diff
@@ -298,4 +298,36 @@ public function passwordReset(Request $request, $db)
         }
 
     }
+
+
+
+    public function deleteAction($request, $db)
+    {
+
+
+        if (! isset($request->user_id)) {
+            throw new Exception("You must be logged in to delete data", 400);
+        }
+        // delete the talk
+        $user_id     = $this->getItemId($request);
+        $user_mapper = new UserMapper($db, $request);
+
+        // note: use the mapper's getUserById as we don't want to throw a not found exception
+        $user = $user_mapper->getUserById($user_id);
+        if (false === $user) {
+            // user isn't there so it's as good as deleted
+            header("Content-Length: 0", null, 204);
```

Comment:
Wether an app shows an error should not be the reason which response an API gives. And when a DELETE is tried multiple times the action can be sucessful the first time and after that you are trying to delete a non-existing resource which should give a 404.

An app that is using that API should know that after deleting a resource a 404 is about as good as a 204 as the resource isn't there (anymore). Whether that's the result of the own action or due to someone else deleting the resource in between shouldn't matter. 

My Personal POV on a philosophical (and much debated) question :wink: